test/hdkeys.js

```diff
@@ -76,6 +76,11 @@ describe('BIP32 compliance', function() {
     publicKey.toString().should.equal(publicKey.xpubkey);
   });
 
+  it('cache for xpubkey works for test vector 1', function() {
+    var pk = HDPrivateKey(vector1_m_private);
+    pk.xpubkey.should.equal(pk.xpubkey);
```

Comment:
Add this line before please: `expect(pk._hdPublicKey).to.be.undefined`; so this test fails before the commit